RevenueCatUI/Templates/V2/Components/Packages/PurchaseButton/PurchaseButtonComponentViewModel.swift

```diff
@@ -24,22 +24,94 @@ class PurchaseButtonComponentViewModel {
     private let offering: Offering
     let stackViewModel: StackComponentViewModel
 
+    private let customWebCheckoutUrl: URL?
+
     init(
+        localizationProvider: LocalizationProvider,
         component: PaywallComponent.PurchaseButtonComponent,
         offering: Offering,
         stackViewModel: StackComponentViewModel
-    ) {
+    ) throws {
         self.component = component
         self.offering = offering
         self.stackViewModel = stackViewModel
+
+        if case let .customWebCheckout(customWebCheckout)? = component.method {
+            self.customWebCheckoutUrl = try localizationProvider
+                .localizedStrings
+                .urlFromLid(customWebCheckout.customUrl.url)
+        } else {
+            self.customWebCheckoutUrl = nil
+        }
+
     }
 
-    var action: PaywallComponent.PurchaseButtonComponent.Action? {
-        return self.component.action
+    var method: PaywallComponent.PurchaseButtonComponent.Method? {
+        return self.component.method ?? self.component.action.flatMap({ action in
+            switch action {
+            case .inAppCheckout:
+                return .inAppCheckout
+            case .webCheckout:
+                return .webCheckout(.init(autoDismiss: true, openMethod: .externalBrowser))
+            case .webProductSelection:
+                return .webProductSelection(.init(autoDismiss: true, openMethod: .externalBrowser))
+            }
+        })
+    }
+
+    typealias LaunchWebCheckout = (url: URL, method: PaywallComponent.ButtonComponent.URLMethod, autoDismiss: Bool)
+
+    func urlForWebCheckout(packageContext: PackageContext?) -> LaunchWebCheckout? {
```

Comment:
I think `PackageContext` can be nonoptional
```suggestion
    func urlForWebCheckout(packageContext: PackageContext) -> LaunchWebCheckout? {
```